syntax = "proto3";

message RoleItem {
    uint64    RoleID  = 1;
    string    Name    = 2;
    int32     Carrer  = 3;
    int32     Level   = 4;
}

message RoleListReq {
    uint64   AccountID    = 1;
    int32    LoginCode    = 2;
    uint32   ServerID     = 3;
}

message RoleListAck {
    uint32    RetCode            = 1;
    uint64    AccountID          = 2;
    repeated  RoleItem RoleList  = 3;
}

message RoleCreateReq {
    uint64   AccountID    = 1;
    string   Name         = 2;
    int32    Carrer       = 3;
    int32    Channel      = 4;
}

message RoleCreateAck {
    uint32    RetCode      = 1;
    uint64    AccountID    = 2;
    uint64    RoleID       = 3;
    string    Name         = 4;
    int32     Carrer       = 5;
}

message RoleDeleteReq {
    uint64    AccountID    = 1;
    uint64    RoleID       = 2;
}

message RoleDeleteAck {
    uint32    RetCode = 1;
}

message RoleLoginReq {
    uint64    AccountID    = 1;
    uint64    RoleID       = 2;
    int32     LangID       = 3; //语言ID
}

message RoleLogoutReq {
    uint64    AccountID = 1;
}

message RoleLogoutAck {
    uint32    RetCode = 1;
}

message RoleOtherLoginNty {
    uint32    RetCode = 1;
}

